src/libraries/System.Net.ServerSentEvents/src/System/Net/ServerSentEvents/SseFormatter.cs

```diff
@@ -119,6 +122,26 @@ private static void FormatSseEvent(
                 bufferWriter.WriteUtf8String(s_newLine);
             }
 
+            if (reconnectionInterval is { } retry)
+            {
+                Debug.Assert(retry >= TimeSpan.Zero);
+
+                long retryMs = (long)retry.TotalMilliseconds;
+
+                bufferWriter.WriteUtf8String("retry: "u8);
+#if NET
+                const int MaxDigits = 20;
+                Span<byte> buffer = stackalloc byte[MaxDigits];
+                bool success = retryMs.TryFormat(buffer, out int bytesWritten, provider: CultureInfo.InvariantCulture);
+                Debug.Assert(success);
+
+                bufferWriter.Write(buffer.Slice(0, bytesWritten));
```

Comment:
Is writing into a stackalloc'd span better than:
```C#
const int MaxDigits = 20;
Span<byte> buffer = bufferWriter.GetSpan(MaxDigits);
bool success = retryMs.TryFormat(buffer, out int bytesWritten, provider: CultureInfo.InvariantCulture);
Debug.Assert(success);
bufferWriter.Advance(bytesWritten);
```
?

Reply:
Good idea, I'll use that instead.

Reply:
Done.